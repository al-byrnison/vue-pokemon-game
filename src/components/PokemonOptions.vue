<template>
  <div class="options-container">
    <ul>
      <li v-for="pokemon in pokemons" :key="pokemon.id" @click="$emit('selectionPokemon', pokemon.id)">
        {{pokemon.name}}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    pokemons: {
      type: Array,
      required: true,
    }
  }
}
</script>

<style scoped>
/* Pokemon Options */
ul {
    list-style-type: none;
  justify-content: center;
  gap: .5rem
}
li {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    cursor: pointer;
    width: 8rem;
    margin: .5rem;;
}

li:hover {
    background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
    display: flex;
    justify-content: center;
}

@media (min-width: 769px) {
  ul {
    display: flex;
  }
}
</style>