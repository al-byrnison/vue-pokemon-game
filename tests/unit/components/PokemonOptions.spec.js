import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions'

import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonOptions Component', () => {
	let wrapper
	beforeEach(() => {
		wrapper = shallowMount(PokemonOptions, {
			props: {
				pokemons,
			},
		})
	})
	test('should snapshot match', () => {
		expect(wrapper.html()).toMatchSnapshot()
	})

	test('should display all 4 options correctly', () => {
		const liTags = wrapper.findAll('li')
		expect(liTags.length).toBe(4)

		expect(liTags[0].text()).toBe('bulbasaur')
		expect(liTags[1].text()).toBe('ivysaur')
		expect(liTags[2].text()).toBe('venusaur')
		expect(liTags[3].text()).toBe('charmander')
	})

	test('should emit "selection" with their respective parameters onClick', () => {
        const [li1, li2, li3, li4] = wrapper.findAll('li')
        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')
        expect(wrapper.emitted('selectionPokemon').length).toBe(4)
        expect(wrapper.emitted('selectionPokemon')[0]).toEqual([1])
        expect(wrapper.emitted('selectionPokemon')[1]).toEqual([2])
        expect(wrapper.emitted('selectionPokemon')[2]).toEqual([3])
        expect(wrapper.emitted('selectionPokemon')[3]).toEqual([4])
    })
})
