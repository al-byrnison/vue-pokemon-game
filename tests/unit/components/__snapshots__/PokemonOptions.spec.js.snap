// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component should snapshot match 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
`;
